functions/snippets/src/main/java/com/example/functions/ImageMagick.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.functions;
+
+// [START functions_imagemagick_setup]
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Feature.Type;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+import com.google.cloud.vision.v1.SafeSearchAnnotation;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class ImageMagick implements BackgroundFunction<GcsEvent> {
+
+  private static Storage storage = StorageOptions.getDefaultInstance().getService();
+  private static final String BLURRED_BUCKET_NAME = System.getenv("BLURRED_BUCKET_NAME");
+  private static final Logger LOGGER = Logger.getLogger(ImageMagick.class.getName());
+  // [END functions_imagemagick_setup]
+
+  // [START functions_imagemagick_analyze]
+  @Override
+  // Blurs uploaded images that are flagged as Adult or Violence.
+  public void accept(GcsEvent gcsEvent, Context context) {
+    BlobInfo blobInfo = BlobInfo.newBuilder(gcsEvent.getBucket(), gcsEvent.getName()).build();
```

Comment:
If users can blindly play w/ our buckets we should talk.

Reply:
They can't - all they can do is download (`read`) the `zombie.jpg` file needed to run this test.

(They have to specify a separate bucket to upload the blurred file to, as per the `README` I added.)